src/3.2/docker-entrypoint.sh

```diff
@@ -38,47 +85,19 @@ if [ "$1" == "neo4j" ]; then
         exit 1
     fi
 
-    setting "dbms.connectors.default_listen_address" "0.0.0.0"
-    setting "dbms.connector.http.listen_address" "0.0.0.0:7474"
-    setting "dbms.connector.https.listen_address" "0.0.0.0:7473"
-    setting "dbms.connector.bolt.listen_address" "0.0.0.0:7687"
-    setting "dbms.mode" "${NEO4J_dbms_mode:-}"
-    setting "dbms.connectors.default_advertised_address" "${NEO4J_dbms_connectors_defaultAdvertisedAddress:-}"
-    setting "ha.server_id" "${NEO4J_ha_serverId:-}"
-    setting "ha.host.data" "${NEO4J_ha_host_data:-}"
-    setting "ha.host.coordination" "${NEO4J_ha_host_coordination:-}"
-    setting "ha.initial_hosts" "${NEO4J_ha_initialHosts:-}"
-    setting "causal_clustering.expected_core_cluster_size" "${NEO4J_causalClustering_expectedCoreClusterSize:-}"
-    setting "causal_clustering.initial_discovery_members" "${NEO4J_causalClustering_initialDiscoveryMembers:-}"
-    setting "causal_clustering.discovery_listen_address" "${NEO4J_causalClustering_discoveryListenAddress:-0.0.0.0:5000}"
-    setting "causal_clustering.discovery_advertised_address" "${NEO4J_causalClustering_discoveryAdvertisedAddress:-$(hostname):5000}"
-    setting "causal_clustering.transaction_listen_address" "${NEO4J_causalClustering_transactionListenAddress:-0.0.0.0:6000}"
-    setting "causal_clustering.transaction_advertised_address" "${NEO4J_causalClustering_transactionAdvertisedAddress:-$(hostname):6000}"
-    setting "causal_clustering.raft_listen_address" "${NEO4J_causalClustering_raftListenAddress:-0.0.0.0:7000}"
-    setting "causal_clustering.raft_advertised_address" "${NEO4J_causalClustering_raftAdvertisedAddress:-$(hostname):7000}"
+    # list env variables with prefix NEO4J_ and create settings from them
+    for i in $( printenv | grep ^NEO4J_ | awk -F'=' '{print $1}' | sort -rn ); do
+        setting=$(echo ${i} | sed 's|^NEO4J_||' | sed 's|_|.|g' | sed 's|\.\.|_|g')
+        value=$(echo ${!i})
+        if grep -q -F "^${setting}=" conf/"${file}"; then
```

Comment:
`conf/${file}` gives an error for unbound variable. I think you want to hardcode this to `conf/neo4j.conf`